frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -208,3 +210,29 @@ export type FirehoseResult<
 export type FirehoseResourcesResult = {
   [key: string]: FirehoseResult<K8sResourceCommon | K8sResourceCommon[]>;
 };
+
+export type Request<R> = {
+  active: boolean;
+  timeout: NodeJS.Timer;
+  inFlight: boolean;
+  data: R;
+  error: any;
+};
+
+export type RequestMap<R> = ImmutableMap<string, Request<R>>;
+export type K8sState = ImmutableMap<string, any>;
+export type UIState = ImmutableMap<string, any>;
+export type DashboardsState = ImmutableMap<string, RequestMap<any>>;
+export type FeatureState = ImmutableMap<string, boolean>;
+
+export type RootState = {
```

Comment:
Are we exposing the Redux store as an API with this change? I'm not sure everything in the store can be an API.

Reply:
I don't think we need to expose the redux store do we?
Can we not expose utils which access redux state but as a consumer you're oblivious to whether the data comes from redux or not.

Reply:
I realized this is in `console-types` which isn't exposed as API.

Reply:
OK thanks, I wanted to confirm that it wasn't a public API.

Reply:
Yeah we really need to organize our SDK types better, because right now it's unclear which ones contribute to public API.